flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/TableConfig.java

```diff
@@ -367,6 +367,14 @@ public void addJobParameter(String key, String value) {
         getConfiguration().set(PipelineOptions.GLOBAL_JOB_PARAMETERS, params);
     }
 
+    /** Returns whether CAST should follow the legacy behaviour or the new one. */
+    @Deprecated
+    public boolean getLegacyCastBehaviour() {
```

Comment:
The `TableConfig` class is an API class that should make frequently used options nicer. I would like to avoid adding new methods there. We should consult the underlying `ReadableConfig` directly.